SQLFrontend/mapdql.cpp

```diff
@@ -1768,9 +1767,19 @@ int main(int argc, char** argv) {
    * The call to linenoise() will block as long as the user types something
    * and presses enter.
    *
-   * The typed string is returned as a malloc() allocated string by
-   * linenoise, so the user needs to free() it. */
-  while ((line = linenoise(prompt.c_str())) != NULL) {
+   * A smart pointer with a custom deleter now manages the typed string; you
+   * do not need to free() it. */
```

Comment:
* The typed string that is malloc() allocated by linenoise() is managed by a smart pointer
* with a custom free() deleter; no need to free this memory explicitly.